lib/bacnet-asn1.js

```diff
@@ -1724,29 +1720,57 @@ var bacapp_decode_context_data = function(buffer, offset, maxApduLen, propertyTa
   };
 };
 
-var encode_bacnet_character_string = function(buffer, value) {
-  buffer.buffer[buffer.offset++] = baEnum.BacnetCharacterStringEncodings.CHARACTER_UTF8;
-  buffer.offset += buffer.buffer.write(value, buffer.offset, undefined, 'utf8');
+var encode_bacnet_character_string = function(buffer, value, encoding) {
+  encoding = encoding || baEnum.BacnetCharacterStringEncodings.CHARACTER_UTF8;
+  buffer.buffer[buffer.offset++] = encoding;
+  if (encoding === baEnum.BacnetCharacterStringEncodings.CHARACTER_UTF8) {
+    buffer.offset += buffer.buffer.write(value, buffer.offset, undefined, 'utf8');
+  } else {
+    // Transcode JS string to the target encoding using iconv-lite
```

Comment:
Do we need those comments here?

`iconv.encode` vs. `Transcode JS string to the target encoding using iconv-lite`